tools/tensorflow/tensor_functions.py

```diff
@@ -0,0 +1,3 @@
+def double(x):
+    y = x * x
+    return y
```

Comment:
This function implementation isn't quite correct; the goal is that the function _itself_ should effectively return `2*x`, but that its implementation should do so by taking the gradient. In other words, the machinery for taking gradients should happen in here, not in `tensor_run.py`. You can take a look at how we do this in `tools/pytorch` or `tools/zygote` to compare.

Reply:
I just pushed commit 5f7aaba070a26cbe412bccbe39e197d2455c1f18 which adds a second function, `square`, that should clarify the intention here.